internal/driver/controller.go

```diff
@@ -430,19 +430,40 @@ func convertRequestCapacityBytes(requestBytes, limitBytes int64) (int64, error)
 		)
 	}
 
-	// if requestBytes is 0, default to max(1Gi, limitBytes>0)
-	// 1. if limitBytes is 0, default to 1Gi
-	// 2. if limitBytes is greater than 1Gi, default to 1Gi
-	// 3. if limitBytes is less than 1Gi, default to limitBytes
+	// if requestBytes is 0 and
+	// 1. if limitBytes == 0, default to 1Gi
+	// 2. if limitBytes >= 1Gi, default to 1Gi
+	// 3. if limitBytes < 1Gi, default to rounded-up 4096 multiple of limitBytes
 	if requestBytes == 0 {
-		if limitBytes > 0 && topolvm.DefaultSize > limitBytes {
-			return limitBytes, nil
+		// if there is no limit or the limit is bigger or equal to the default, use the default
+		if limitBytes == 0 || limitBytes >= topolvm.DefaultSize {
+			return topolvm.DefaultSize, nil
+		}
+
+		return roundUp(limitBytes, topolvm.MinimumSectorSize), nil
```

Comment:
As we can see from the test for `convertRequestCapacityBytes(0, topolvm.MinimumSectorSize-1)` ), in this case,  the return value might exceed the limit, which violates the CSI spec. I think we should round **down** the limit and

1. if zero return error,
2. otherwise return it.

Reply:
Instead of changing to rounding down, we can also decide to simply reject this case and return an error if the limit<MinimumSectorSize. I believe this is actually a much cleaner behavior. Rounding down will provision less capacity than a PVC requests and that might become a problem, while overprovisioning never becomes problematic for the Application. WDYT?